core/primitives/src/views.rs

```diff
@@ -2378,29 +2381,54 @@ impl TryFrom<ReceiptView> for Receipt {
                     input_data_ids,
                     actions,
                     is_promise_yield,
+                    refund_to,
                 } => {
-                    let action_receipt = ActionReceipt {
-                        signer_id,
-                        signer_public_key,
-                        gas_price,
-                        output_data_receivers: output_data_receivers
-                            .into_iter()
-                            .map(|data_receiver_view| DataReceiver {
-                                data_id: data_receiver_view.data_id,
-                                receiver_id: data_receiver_view.receiver_id,
-                            })
-                            .collect(),
-                        input_data_ids: input_data_ids.into_iter().map(Into::into).collect(),
-                        actions: actions
-                            .into_iter()
-                            .map(TryInto::try_into)
-                            .collect::<Result<Vec<_>, _>>()?,
-                    };
-
-                    if is_promise_yield {
-                        ReceiptEnum::PromiseYield(action_receipt)
+                    let output_data_receivers: Vec<_> = output_data_receivers
+                        .into_iter()
+                        .map(|data_receiver_view| DataReceiver {
+                            data_id: data_receiver_view.data_id,
+                            receiver_id: data_receiver_view.receiver_id,
+                        })
+                        .collect();
+                    let input_data_ids: Vec<CryptoHash> =
+                        input_data_ids.into_iter().map(Into::into).collect();
+                    let actions = actions
+                        .into_iter()
+                        .map(TryInto::try_into)
+                        .collect::<Result<Vec<_>, _>>()?;
+                    // Note that this is not consistent with how `new_receipts` are
+                    // created by the runtime - there we always create ActionReceiptV2.
+                    // ActionReceiptV2 without refund_to becomes V1 after a roundtrip
+                    // through views. This will be fixed with #14709.
```

Comment:
I would still add a literal reference to `ProtocolFeature::DeterministicAccountIds` so that somebody grepping through this feature looking for code to remove would also find this code as well.

I believe such a grep is exactly what @shreyan-gupta has been doing for PRs like https://github.com/near/nearcore/pull/14408

Reply:
Inlining `DeterministicAccountIds` wouldn't solve the problem because we still use old `ActionReceipt` as part of `Receipt::from_tx`, so it will remain inconsistent. That is why I've created #14709 to properly fix that and use `ActionReceiptV2` everywhere.